include/zephyr/arch/riscv/arch_inlines.h

```diff
@@ -17,6 +17,9 @@ static ALWAYS_INLINE uint32_t arch_proc_id(void)
 
 #ifdef CONFIG_SMP
 	__asm__ volatile("csrr %0, mhartid" : "=r" (hartid));
+#if CONFIG_RV_BASE_CPU > 0
+		hartid -= CONFIG_RV_BASE_CPU;
+#endif
```

Comment:
If `CONFIG_RV_BASE_CPU` defaults to `0` anyway, I think we can skip this `if` directive here. It clutters the code a bit, and the compiler will optimize the 0-subtraction anyway.

Reply:
Hi @fkokosinski,
Thanks for pointing this out. done.